improver_tests/nbhood/recursive_filter/test_RecursiveFilter.py

```diff
@@ -278,77 +290,122 @@ def test_with_separate_mask_cube_and_nan(self):
-        msg = "The points of the x spatial dimension of the smoothing coefficients"
+        msg = (
+            "The smoothing coefficients x dimension does not have the "
+            "expected length or values"
+        )
         with self.assertRaisesRegex(ValueError, msg):
-            RecursiveFilter(edge_width=1)._validate_smoothing_coefficients(
-                self.cube, self.smoothing_coefficients_cube_wrong_x_points
+            RecursiveFilter(edge_width=1)._validate_and_pad_coefficients(
+                self.cube, self.smoothing_coefficients_wrong_points
             )
 
-    def test_smoothing_coefficients_mismatched_y_points(self):
-        """Test that an error is raised if the y smoothing_coefficients_cube
-        has mismatched coordinate points compared to the data cube."""
-        msg = "The points of the y spatial dimension of the smoothing coefficients"
+    def test_smoothing_coefficients_exceed_max(self):
+        """Test that an error is raised if any smoothing coefficient value
+        exceeds the allowed maximum of 0.5."""
+        self.smoothing_coefficients[0].data += 1.0
+        msg = (
+            "All smoothing_coefficient values must be less than 0.5. "
+            "A large smoothing_coefficient value leads to poor "
+            "conservation of probabilities"
+        )
         with self.assertRaisesRegex(ValueError, msg):
-            RecursiveFilter(edge_width=1)._validate_smoothing_coefficients(
-                self.cube, self.smoothing_coefficients_cube_wrong_y_points
+            RecursiveFilter(edge_width=1)._validate_and_pad_coefficients(
+                self.cube, self.smoothing_coefficients
             )
 
-
-class Test__set_smoothing_coefficients(Test_RecursiveFilter):
-
-    """Test the _set_smoothing_coefficients method"""
-
-    def test_smoothing_coefficients_cube(self):
-        """Test that the returned smoothing_coefficients array has the expected
-        result when smoothing_coefficients_cube is not None."""
-        result = RecursiveFilter(edge_width=1)._set_smoothing_coefficients(
-            self.smoothing_coefficients_cube_x
+    def test_padding_default(self):
+        """Test that the returned smoothing_coefficients array is padded as
+        expected with the default edge_width."""
+        expected_shape = (65, 64)
+        expected_result = np.full(expected_shape, 0.5)
+        result, _ = RecursiveFilter()._validate_and_pad_coefficients(
+            self.cube, self.smoothing_coefficients
         )
-        expected_result = 0.5
         self.assertIsInstance(result.data, np.ndarray)
-        self.assertEqual(result.data[0][2], expected_result)
-        # Check shape: Array should be padded with 4 extra rows/columns
+        self.assertArrayEqual(result.data, expected_result)
+        # Check shape: Array should be padded with 15 * 4 extra rows/columns
+        self.assertEqual(result.shape, expected_shape)
```

Comment:
Can you specify that this is because of the edge width used (which is ...?)

Reply:
Done.